# Local costmap parameters
# Used for local planning and collision avoidance

local_costmap:
  global_frame: odom
  robot_base_frame: base_footprint
  update_frequency: 5.0
  publish_frequency: 2.0
  static_map: false
  rolling_window: true
  width: 4.0
  height: 4.0
  resolution: 0.05
  origin_x: -2.0
  origin_y: -2.0
  
  # Layer defaults
  default_value: 0
  unknown_cost_value: -1
  
  # Plugins for local costmap
  plugins:
    - {name: obstacle_layer, type: 'costmap_2d::VoxelLayer'}
    - {name: inflation_layer, type: 'costmap_2d::InflationLayer'}
  
  # Obstacle layer specific to local costmap
  obstacle_layer:
    enabled: true
    observation_sources: laser
    laser: 
      sensor_frame: lidar_link
      data_type: LaserScan
      topic: /scan
      marking: true
      clearing: true
      max_obstacle_height: 0.5
      min_obstacle_height: 0.0
      expected_update_rate: 5.0
      observation_persistence: 0.0
      inf_is_valid: true
      track_unknown_space: false
    footprint_clearing_enabled: true
    combination_method: 1
  
  # Inflation layer for local costmap
  inflation_layer:
    enabled: true
    cost_scaling_factor: 10.0
    inflation_radius: 0.20

